### ISSUE TITLE
Add measures to prevent overfitting

### ISSUE THREAD

**Issue by pranshu-raj-211** [pranshu-raj-211/mnist_GAN#4]
The model might be overfitting on the data - to mitigate this I can add a bit of noise to the labels and try other best practices used with GANs, modifying the loss function to make it more robust and so on.
